decoder/completion_resolve.go

```diff
@@ -0,0 +1,13 @@
+package decoder
+
+import (
+	"context"
+)
+
+func (d *Decoder) ResolveCandidate(ctx context.Context, unresolvedCandidate UnresolvedCandidate) (*ResolvedCandidate, error) {
```

Comment:
Nitpick: Do you mind adding a comment to the method, as it is part of the public API?

Reply:
@dbanck I think this still remains unresolved - and would lead to a crash if passed `unresolvedCandidate.ResolveHook` is `nil`.